operator/pkg/helmreconciler/common.go

```diff
@@ -125,5 +126,77 @@ func applyOverlay(current, overlay runtime.Object) error {
 	if err != nil {
 		return err
 	}
-	return runtime.DecodeInto(unstructured.UnstructuredJSONScheme, merged, current)
+	// save any immutable values
+	clusterIP := getPath(current, name.ServiceStr, util.PathFromString("spec.clusterIP"))
+
+	err = runtime.DecodeInto(unstructured.UnstructuredJSONScheme, merged, current)
```

Comment:
Why not return early? if `runtime.DecodeInto` fails, the subsequent steps will also fail?

Reply:
fixed, thanks